nvflare/apis/impl/study_manager.py

```diff
@@ -33,12 +33,18 @@ def custom_json_encoder(obj):
 
 
 class StudyManager(StudyManagerSpec):
+    STORAGE_KEY = "std_mgr"
+
     def __init__(self, storage: StorageSpec):
+        self._name = self.__class__.__name__
+        self.logger = logging.getLogger(self._name)
         self._storage = storage
-        self._tmp_dir = tempfile.TemporaryDirectory()
-        self._existing_studies = dict()
+        try:
+            self._existing_studies = json.loads(self._storage.get_data(StudyManager.STORAGE_KEY).decode("utf-8"))
+        except BaseException:
+            self._existing_studies = list()
 
-    def create_study(self, study: Study) -> Tuple[Optional[Study], str]:
+    def add_study(self, study: Study) -> Tuple[Optional[Study], str]:
```

Comment:
maybe we want to also update the docstrings.

Other than this,
For the method docstrings after Returns: there should be a new line.

Reply:
Or should we change the spec name from "add" to "create"?